atd-vzv/src/views/summary/CrashesByTimeOfDay.js

```diff
@@ -96,13 +101,77 @@ const CrashesByTimeOfDay = () => {
       return queryUrl;
     };
 
-    // Wait for crashType to be passed up from setCrashType component,
-    // then fetch records for selected year
-    if (crashType.queryStringCrash)
-      axios.get(getFatalitiesByYearsAgoUrl()).then((res) => {
-        setHeatmapData(calculateHourBlockTotals(res.data));
+    axios.get(getFatalitiesByYearsAgoUrl()).then((res) => {
+      const formattedData = calculateHourBlockTotals(res.data);
+      setHeatmapData(formattedData);
+    });
+  }, [activeTab, crashType]);
+
+  // Query to find maximum day total per crash type
+  useEffect(() => {
+    if (maxForLegend) return;
+
+    const maxQuery = `
+    SELECT date_extract_dow(crash_date) as day, date_extract_hh(crash_date) as hour, date_extract_y(crash_date) as year, SUM(death_cnt) as death, SUM(sus_serious_injry_cnt) as serious, serious + death as all 
+    WHERE crash_date BETWEEN '${dataStartDate.format(
+      "YYYY-MM-DD"
+    )}' and '${summaryCurrentYearEndDate}' 
+    GROUP BY day, hour, year 
+    ORDER BY year 
+    |> 
+    SELECT max(death) as fatalities, max(serious) as seriousInjuries, max(all) as fatalitiesAndSeriousInjuries
+    `;
```

Comment:
Very cool to see that chain operator `|>` learning something cool!

Reply:
Yes! I was happy to find a use for this hidden gem.

Reply:
✨